sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiWriteUnshardedRecords.java

```diff
@@ -643,11 +650,20 @@ long flush(
             contexts -> {
               AppendRowsContext failedContext =
                   Preconditions.checkStateNotNull(Iterables.getFirst(contexts, null));
+              Instant operationEndTime = Instant.now();
```

Comment:
We have this pattern all over the place, of grabbing the start time, updating a metric, grabbing error code, etc.

Is it possible to repackage this into a reusable utility method for clarity?

Reply:
Repacked updating the `RpcRequests` and `RpcLatency` metrics into utility methods since that are reused across all write methods. These updates are now handled by `reportSuccessfulRpcMetrics` and `reportFailedRPCMetrics`.

It's not straightforward to refactor updates `AppendRowsRowStatus` into a utility. When we update this metric we need to know the number of rows that were failed/retried/succeeded so it's more embedded in the code.